pymobiledevice3/services/crash_reports.py

```diff
@@ -127,28 +128,36 @@ def watch(self, name: str = None, raw: bool = False) -> Generator[str, None, Non
                 else:
                     yield crash_report
 
-    def get_new_sysdiagnose(self, out: str, erase: bool = True) -> None:
+    def get_new_sysdiagnose(self, out: str, erase: bool = True, *, timeout: Optional[float] = None) -> None:
         """
         Monitor the creation of a newly created sysdiagnose archive and pull it
         :param out: filename
         :param erase: remove after pulling
+        :keyword timeout: Maximum time in seconds to wait for the completion of sysdiagnose archive
+            If None (default), waits indefinitely
         """
-        sysdiagnose_filename = self._get_new_sysdiagnose_filename()
+        if timeout is None:
+            timeout = float('inf')
+        timeout = time.time() + timeout
+        sysdiagnose_filename = self._get_new_sysdiagnose_filename(timeout)
         self.logger.info('sysdiagnose tarball creation has been started')
-        self._wait_for_sysdiagnose_to_finish()
+        self._wait_for_sysdiagnose_to_finish(timeout)
         self.pull(out, entry=sysdiagnose_filename, erase=erase)
 
     @staticmethod
     def _sysdiagnose_complete_syslog_match(message: str) -> bool:
         return message == 'sysdiagnose (full) complete' or 'Sysdiagnose completed' in message
 
-    def _wait_for_sysdiagnose_to_finish(self) -> None:
+    def _wait_for_sysdiagnose_to_finish(self, timeout: float) -> None:
         with OsTraceService(self.lockdown) as os_trace:
             for entry in os_trace.syslog():
                 if CrashReportsManager._sysdiagnose_complete_syslog_match(entry.message):
                     break
 
-    def _get_new_sysdiagnose_filename(self) -> str:
+                if time.time() > timeout:
```

Comment:
should be something life `time.time() - start > timeout`

Reply:
addressed this by using `end_time` variable and added `_check_timeout` method.